zbar/qrcode/qrdectxt.c

```diff
@@ -185,10 +189,15 @@ int qr_code_data_list_extract_text(const qr_code_data_list *_qrlist,
       else sa_text[sa_ntext++]=(char)(fnc1_2ai-100);
     }
     eci=-1;
+#ifndef ALL_ENCODINGS
+    enc_list[0]=utf8_cd;
```

Comment:
This writes after the stack pointer enc_list... maybe into ```eci_cd``` depends on the compiler mood.

You want to reserve a list of 1 element size of enc_list... otherwise you end up writing to some random piece of stack.